test/test_main.py

```diff
@@ -381,10 +381,7 @@ def _read_and_clear_state(
             class ThisCharmEvents(MyCharmEvents):
                 pass
 
-            class _HasStored(typing.Protocol):
-                _stored: ops.StoredState
-
-            class Charm(self.charm_module.Charm, _HasStored):  # type: ignore
```

Comment:
I thought that I removed the attribute **declaration** only.

The test still passes, so maybe I was wrong why this Protocol was added in the first place?

I guessed that the protocol was added because we don't actually know anything about the Charm as it was loaded dynamically and a kludge was needed, e.g. because `._stored` was accessed right below, or some such.

Let's cover this during the stand-up, I'm happy to learn the back story.

Reply:
> I thought that I removed the attribute **declaration** only.

Ah, you did, sorry. I misread before.

> The test still passes, so maybe I was wrong why this Protocol was added in the first place?

I'm not sure why it exists, or why things are good without it.

> Let's cover this during the stand-up, I'm happy to learn the back story.

Hopefully, Ben knows :)

Reply:
> I thought that I removed the attribute **declaration** only.

Ah, you did, sorry. I misread before.

> The test still passes, so maybe I was wrong why this Protocol was added in the first place?

I'm not sure why it exists, or why things are good without it.

> Let's cover this during the stand-up, I'm happy to learn the back story.

Hopefully, Ben knows :)